torch/csrc/jit/codegen/cuda/scheduler/utils.h

```diff
@@ -492,6 +505,51 @@ struct TORCH_CUDA_CU_API BoundedDirectionalTransformPropagator {
       Options options);
 };
 
+// Schedulers typically start by merging some axes together then splitting,
+// and propagating those transformations through the dag. What we want to
+// understand is if these merges can be supported through view operations.
+// For example it could be problematic to support a reduction fusion:
+//
+// tv0[2, 3, 4]
+// tv1 = sum(tv0, {1, 2})
+// tv2 = view(tv0, {6, 4})
+//
+// Since the first step of the reduction scheduler would be tv1->merge(1, 2).
+// If we tried to propagate this transformation through the view it would make
+// the view invalid. If we tried to propagate the view through the reduction,
+// it would attempt to merge a reduction and non-reduction dimension. So for
+// these types of fusions we would like to understand that the view considers
+// axis 1 and 2 of tv1 as "non-separable" axes.
+//
+// If IterDomains are disjoint in the returned set, then they are considered
+// "separable".
+// Warning: This pass generates the IdGraphs, not intended for use at runtime.
+TORCH_CUDA_CU_API DisjointSets<IterDomain*> disjointViewSets(Fusion* fusion);
+
+// Return if all trasnformations in all views match.
+// TODO: Should this be moved to registry.cpp/.h?
+// Warning: This pass generates the IdGraphs, not intended for use at runtime.
+TORCH_CUDA_CU_API bool allMatchingViews(Fusion* fusion);
+
+// Return if there are any view's that have dependencies on other view
+// operations.
+TORCH_CUDA_CU_API bool hasDependentViews(Fusion* fusion);
```

Comment:
Is this used in any scheduler?

Reply:
No, I think this was before I decided either we have a producer-consumer path to a reference from all views or all views have to match, meaning they can't be dependent on eachother. Will remove.